sdks/python/apache_beam/io/watermark_estimators.py

```diff
@@ -0,0 +1,101 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""A collection of WatermarkEstimator implementations that SplittableDoFns
+can use."""
+
+from __future__ import absolute_import
+
+from apache_beam.io.iobase import WatermarkEstimator
+from apache_beam.utils.timestamp import Timestamp
+
+
+class MonotonicWatermarkEstimator(WatermarkEstimator):
+  """A WatermarkEstimator which assumes that timestamps of all ouput records
+  are increasing monotonically.
+  """
+  def __init__(self, timestamp):
+    """For a new tracking <element, restriction> pair, the initial value
+    should be None. When resuming from residual, the initial timestamp should
+    be the same timestamp as the estimated watermark from primary.
+    """
+    self._watermark = timestamp
+
+  def observe_timestamp(self, timestamp):
+    if self._watermark is None:
+      self._watermark = timestamp
+    else:
+      if timestamp < self._watermark:
+        raise ValueError('A MonotonicWatermarkEstimator expects output '
+                         'timestamp to be increasing monotonically.')
```

Comment:
```suggestion
                         'watermark to be increasing monotonically.')
```

Reply:
The `MonotonicWatermarkEstimator.observe_timestamp()` expects the timestamp of output_record. For me, conceptually it's a timestamp.

Reply:
I agree, it observes monotonically increasing timestamps and computes, based on those, a watermark.